### PR TITLE
Link back to Cargo.toml section.

### PR DESCRIPTION
**Pull request by Dygear**
This is mostly so that when someone comes into this project sideways (they are linked into the document directly, into a section that has a code example) they have context so they know what they need to build the project correctly.

### CODE REVIEW

**Comment by Dygear** [comment]
There is one area that I would like to discuss. The Cargo.toml file that these sections all link back to is [this file here](https://github.com/embassy-rs/embassy/blob/main/docs/modules/ROOT/examples/basic/Cargo.toml).

```TOML
[dependencies]
embassy-executor = { version = "0.5.0", path = "../../../../../embassy-executor", features = ["defmt", "integrated-timers", "arch-cortex-m", "executor-thread"] }
embassy-time = { version = "0.3.0", path = "../../../../../embassy-time", features = ["defmt"] }
embassy-nrf = { version = "0.1.0", path = "../../../../../embassy-nrf", features = ["defmt", "nrf52840", "time-driver-rtc1", "gpiote"] }
```

Is there a value to having this path'd internally? Would it perhaps confuse newer Rust people? While it's nice that this file links into the main / trunk, could we perhaps move to the more standard setup?

```TOML
[dependencies]
embassy-executor = { version = "0.5.0", features = ["defmt", "integrated-timers", "arch-cortex-m", "executor-thread"] }
embassy-time = { version = "0.3.0", features = ["defmt"] }
embassy-nrf = { version = "0.1.0", features = ["defmt", "nrf52840", "time-driver-rtc1", "gpiote"] }
```

File: docs/modules/ROOT/pages/sharing_peripherals.adoc [hunk 1]
```diff
@@ -8,6 +8,7 @@ The following examples shows different ways to use the on-board LED on a Raspber
 
 Using mutual exclusion is the simplest way to share a peripheral.
 
+TIP: The depenencies you need to run this example link:/book/dev/basic_application.html#_the_cargo_toml[can be found here].
```
Comment (plaes) [inline docs/modules/ROOT/pages/sharing_peripherals.adoc:11]: Repeated typo: "depenencies". 

Also, I would rephrase the sentence to use more generic wording:
"Dependencies needed to run this example ...

**Comment by lulf** [comment]
> There is one area that I would like to discuss. The Cargo.toml file that these sections all link back to is [this file here](https://github.com/embassy-rs/embassy/blob/main/docs/modules/ROOT/examples/basic/Cargo.toml).
> 
> ```toml
> [dependencies]
> embassy-executor = { version = "0.5.0", path = "../../../../../embassy-executor", features = ["defmt", "integrated-timers", "arch-cortex-m", "executor-thread"] }
> embassy-time = { version = "0.3.0", path = "../../../../../embassy-time", features = ["defmt"] }
> embassy-nrf = { version = "0.1.0", path = "../../../../../embassy-nrf", features = ["defmt", "nrf52840", "time-driver-rtc1", "gpiote"] }
> ```
> 
> Is there a value to having this path'd internally? Would it perhaps confuse newer Rust people? While it's nice that this file links into the main / trunk, could we perhaps move to the more standard setup?
> 
> ```toml
> [dependencies]
> embassy-executor = { version = "0.5.0", features = ["defmt", "integrated-timers", "arch-cortex-m", "executor-thread"] }
> embassy-time = { version = "0.3.0", features = ["defmt"] }
> embassy-nrf = { version = "0.1.0", features = ["defmt", "nrf52840", "time-driver-rtc1", "gpiote"] }
> ```

This is necessary for the examples to build in CI and be in sync with the repo versions.

File: docs/modules/ROOT/pages/sharing_peripherals.adoc (hunk 1 above)
Comment (Dygear) [reply docs/modules/ROOT/pages/sharing_peripherals.adoc]: Thanks! Fixed!

**Comment by Dygear** [comment]
Thank you for the review @plaes! I've fixed the typos.

@lulf Yeah, I had a feeling that would be the case. Is there a chance that it would be acceptable to hide that internal bit of complexity away from the documentation?

**Review by Dirbaio — APPROVED** [review]
